Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -290,6 +290,91 @@ For white-box source code testing, you'll need a setup similar to the developer'
 
 During **black-box testing**, you won't have access to the original form of the source code. You'll usually have the application package in [Android's APK format](https://en.wikipedia.org/wiki/Android_application_package "Android application package"), which can be installed on an Android device or reverse engineered as explained in the section "Disassembling and Decompiling".
 
+#### Basic Information Gathering
+
+As discussed in previous sections, an Android application can consist of both Java bytecode and native code. To start with analysis of an application, each type of code may require different approach and tools. In this section, we will learn about some approaches and tools for collecting basic information about a given Android application using static analysis.
+
+##### Retrieving Strings
+
+While performing any kind of binary analysis, strings can be considered as one of the most valuable starting point. Strings provide a context to a part of the binary being analyzed. For example, an error log string like "Data encryption failed." gives us a hint that the adjoining code is responsible for performing some kind of encryption operation.
+
+###### Java
+
+As we already know, all the Java bytecode of an Android application is compiled into a dex file. Each dex file contains a [string identifiers list](https://source.android.com/devices/tech/dalvik/dex-format#file-layout "string identifiers list") (strings_ids). Each string used in the code is given an identifier number, and internally this identifier number is used whenever a string is referred. This list contains all the string identifiers used in the binary, including internal naming (e.g, type descriptors) or constant objects referred by the code (e.g hard coded strings). Knowing this, we can simply dump this list to get all the strings used in the Java bytecode in a dex file. Fortunately, there are plenty of tools already available for dumping strings form Java bytecode and one can choose depending on the needs. In this section we will use Ghidra (GUI based), and [Dextra](http://newandroidbook.com/tools/dextra.html "Dextra") (CLI based).
```

Comment:
```suggestion
As we already know, all the Java bytecode of an Android application is compiled into a dex file. Each dex file contains a [string identifiers list](https://source.android.com/devices/tech/dalvik/dex-format#file-layout "string identifiers list") (strings_ids). Each string used in the code is given an identifier number, and internally this identifier number is used whenever a string is referred. This list contains all the string identifiers used in the binary, including internal naming (e.g, type descriptors) or constant objects referred by the code (e.g hard coded strings). Knowing this, we can simply dump this list to get all the strings used in the Java bytecode that is part of the dex file. Fortunately, there are plenty of tools already available for dumping strings form Java bytecode and one can choose depending on the needs. In this section we will use Ghidra (GUI based), and [Dextra](http://newandroidbook.com/tools/dextra.html "Dextra") (CLI based).
```

Reply:
Dextra seems to be 3 years old and I also couldn't find the download link. Is this tool still maintained and where are the sources of Dextra? Was the tool working reliable for you?

Reply:
Dextra is by Jonathan Levin, he do not release the source code of his tools.  the tool is good. Hard to say how actively it is maintained, but it still works for me and I use it quite often.